runner/runner.go

```diff
@@ -53,25 +53,10 @@ func NewCommand(argv []string, env map[string]string, timeout time.Duration, sna
 	return &Command{Argv: argv, EnvVars: env, Timeout: timeout, SnapshotId: snapshotId}
 }
 
-type Runner interface {
-	// Run instructs the Runner to run cmd. It returns its status and any errors.
-	// Run may:
-	// enqueue cmd (leading to state PENDING)
-	// run cmd immediately (leading to state RUNNING)
-	// check if cmd is well-formed, and reject it if not (leading to state FAILED)
-	// wait a very short period of time for cmd to finish
-	// Run may not wait indefinitely for cmd to finish. This is an async API.
-	Run(cmd *Command) (ProcessStatus, error)
-
-	// Status checks the status of run.
-	Status(run RunId) (ProcessStatus, error)
-
-	// Current status of all runs, running and finished, excepting any Erase()'s runs.
-	StatusAll() ([]ProcessStatus, error)
-
-	// Kill the given run.
-	Abort(run RunId) (ProcessStatus, error)
-
-	// Prunes the run history so StatusAll() can return a reasonable number of runs.
-	Erase(run RunId) error
-}
+// TODO(dbentley): do we want to have Runner interface?
+// It combines a few things that are related, but different responsibilities.
+// Perhaps we could rename it something like "RunService", to make clear it's both ends?
+// type Runner interface {
+// 	Controller
+// 	LegacyStatuses
+// }
```

Comment:
I'd want something like a RunService interface plus a good default impl or two so any interaction between Controller & Statuses can be encapsulated at the right level. There is no interaction now but there could be...